modules/index-service/src/main/java/org/opencastproject/index/service/impl/util/PublicationEventUpdateHandler.java

```diff
@@ -119,7 +120,7 @@ public void execute(AssetManagerItem messageItem) {
   }
 
   private void updatePublications(MediaPackage mediaPackage) {
-    for (Publication publication : mediaPackage.getPublications()) {
+    for (Publication publication : new ArrayList<>(Arrays.asList(mediaPackage.getPublications()))) {
```

Comment:
There are a few ways to make this shorter and probably more efficient. I think my fav. so far is

```suggestion
    for (Publication publication : mediaPackage.getPublications().clone()) {
```

Reply:
Double better, `MediaPackageImpl` does this when you `.getPublications()` ;)
```
return mlist(elements).bind(presentations).value().toArray(new Publication[0]);
```

So just use the raw `Publication[]` you're getting back.